### DISCUSSION TITLE
usePaginationQuery double request

### DISCUSSION THREAD

**Discussion by ronaldruzicka** [TanStack/query#1287] [Q&A] [answered]
Hi, I have a question regarding `usePaginationQuery`

Why is it firing two requests when I click on the prev/next page buttons?
In network you can see that the first request is with the old/previous data and the second is with the new/next page data.
Isn't the first request for old data unnecessary?

Here is code sandbox: https://codesandbox.io/s/sweet-brattain-5fkyf

**Comment by TkDodo** [comment, marked as answer]
I think what you are probably seeing is that you focus the window at the same time as clicking the next button. This will first trigger the `refetchOnWindowFocus`, which will fetch the current page, and then click the button and go to the next page. It happens quite frequently if you switch between the devtools and the actual app.

So try clicking somewhere else on the page, observe the focus refetch, and then click on the next button - it seems to work just fine :)

**Reply by ronaldruzicka** [reply]
Oooh, ok that makes sense, thanks! 🙂
